warehouse/ingest-core/src/main/java/datawave/ingest/data/config/CachedFieldConfigHelper.java

```diff
@@ -0,0 +1,208 @@
+        this.resultCache = new LruCache<>(limit);
+        this.limit = limit;
+        this.debugLimitsEnabled = debugLimitEnabled;
+        this.debugFieldUnique = new HashSet<>();
+        this.debugFieldComputes = new AtomicLong();
+
+        if (debugLimitEnabled) {
+            this.debugStateExecutor = Executors.newSingleThreadScheduledExecutor(
+            // @formatter:off
+                new ThreadFactoryBuilder()
+                    .setPriority(NORM_PRIORITY)
+                    .setDaemon(true)
+                    .setNameFormat("CachedFieldConfigHelper.DebugState")
+                    .build()
+                // formatter:off
+            );
+            this.debugStateExecutor.scheduleAtFixedRate(this::debugLogState, DEFAULT_DEBUG_STATE_SECS, DEFAULT_DEBUG_STATE_SECS, SECONDS);
+        } else {
+            this.debugStateExecutor = null;
+        }
+    }
+
+    @Override
+    public boolean isStoredField(String fieldName) {
+        return getFieldResult(AttributeType.STORED_FIELD, fieldName, underlyingHelper::isStoredField);
+    }
+
+    @Override
+    public boolean isIndexedField(String fieldName) {
+        return getFieldResult(AttributeType.INDEXED_FIELD, fieldName, underlyingHelper::isIndexedField);
+    }
+
+    @Override
+    public boolean isIndexOnlyField(String fieldName) {
+        return getFieldResult(AttributeType.INDEX_ONLY_FIELD, fieldName, underlyingHelper::isIndexOnlyField);
+    }
+
+    @Override
+    public boolean isReverseIndexedField(String fieldName) {
+        return getFieldResult(AttributeType.REVERSE_INDEXED_FIELD, fieldName, underlyingHelper::isReverseIndexedField);
+    }
+
+    @Override
+    public boolean isTokenizedField(String fieldName) {
+        return getFieldResult(AttributeType.TOKENIZED_FIELD, fieldName, underlyingHelper::isTokenizedField);
+    }
+
+    @Override
+    public boolean isReverseTokenizedField(String fieldName) {
+        return getFieldResult(AttributeType.REVERSE_TOKENIZED_FIELD, fieldName, underlyingHelper::isReverseTokenizedField);
+    }
+
+    @VisibleForTesting
+    boolean getFieldResult(AttributeType attributeType, String fieldName, Predicate<String> fn) {
+        CachedEntry ce = !debugLimitsEnabled ?
+            resultCache.computeIfAbsent(fieldName, CachedEntry::new) :
+            resultCache.computeIfAbsent(fieldName, this::debugCachedEntryCreation);
+        return ce.get(attributeType).getResultOrEvaluate(fn);
+    }
+
+    @VisibleForTesting
+    boolean hasLimitExceeded() {
+        return resultCache.hasLimitExceeded();
+    }
+
+    private CachedEntry debugCachedEntryCreation(String fieldName) {
+        debugFieldComputes.incrementAndGet();
+        debugFieldUnique.add(fieldName);
+        return new CachedEntry(fieldName);
+    }
+
+    private void debugLogState() {
+        if (resultCache.hasLimitExceeded()) {
+            log.info("Field cache LRU limit exceeded [limit={}, debug={}, size={}, uniq={}]",
```

Comment:
Since this feature is "debugLogState" I was expecting this to be debug level logging. Not a deal breaker but I'd vote to make it debug.

Reply:
I can update - I set the level this way so it would show up in logs without having to also adjust logger/levels.